### ISSUE TITLE
Bug resulting in crazy barrier backshore alignment

### ISSUE THREAD

**Issue by RegMeasures** [RegMeasures/HapuaModel#68]
![image](https://user-images.githubusercontent.com/17330572/77113806-527d4e00-6a90-11ea-859a-0b2d15b86600.png)

**Comment by RegMeasures** [comment]
Not just a plotting issue - confirmed by checking data in netCDF file.

In test model SimplifiedSynthetic_v2:
* Landward incursion at X = +20 is rapid but not instantaneous and occurs on 5-6 Feb 2002.
* Seaward excursion at X = -60 is gradual and occurs over July 2002

**Comment by RegMeasures** [comment]
![image](https://user-images.githubusercontent.com/17330572/77115145-a9842280-6a92-11ea-95fd-46b69922e402.png)
```
INFO   : Time = 2002-07-05 00:05:00, MorDt = 3840.0s
INFO   : Closure occured in lagoon at X=-960
INFO   : Outlet intersects lagoon at X = -60.0: adding channel width into lagoon
INFO   : Barrier completely eroded at X = -60
INFO   : Outlet truncation due to breach at X = -60.000000
INFO   : Time = 2002-07-05 01:09:00, MorDt = 3840.0s
INFO   : Outlet channel closed by wave washover at X = -60.000000
INFO   : Barrier completely eroded at X = -60
INFO   : Breach/creation of new outlet channel at X = -60.000000
INFO   : Time = 2002-07-05 02:13:00, MorDt = 3840.0s
INFO   : Closure occured in lagoon at X=-960
INFO   : Outlet intersects lagoon at X = -60.0: adding channel width into lagoon
INFO   : Barrier completely eroded at X = -60
INFO   : Outlet truncation due to breach at X = -60.000000
INFO   : Time = 2002-07-05 03:17:00, MorDt = 3840.0s
INFO   : Outlet channel closed by wave washover at X = -60.000000
INFO   : Barrier completely eroded at X = -60
INFO   : Breach/creation of new outlet channel at X = -60.000000
INFO   : Time = 2002-07-05 04:21:00, MorDt = 3840.0s
```
etc.

**Comment by RegMeasures** [comment]
```INFO   : Closure occured in lagoon at X=-960
INFO   : Outlet intersects lagoon at X = -60.0: adding channel width into lagoon
INFO   : Barrier completely eroded at X = -60
INFO   : Outlet truncation due to breach at X = -60.000000
INFO   : Time = 2002-07-05 01:09:00, MorDt = 3840.0s
INFO   : Outlet channel closed by wave washover at X = -60.000000
INFO   : Barrier completely eroded at X = -60
INFO   : Breach/creation of new outlet channel at X = -60.000000
INFO   : Time = 2002-07-05 02:13:00, MorDt = 3840.0s
INFO   : Closure occured in lagoon at X=-960
INFO   : Outlet intersects lagoon at X = -60.0: adding channel width into lagoon
INFO   : Barrier completely eroded at X = -60
INFO   : Outlet truncation due to breach at X = -60.000000
INFO   : Time = 2002-07-05 03:17:00, MorDt = 3840.0s
INFO   : Outlet channel closed by wave washover at X = -60.000000
INFO   : Barrier completely eroded at X = -60
INFO   : Breach/creation of new outlet channel at X = -60.000000
INFO   : Time = 2002-07-05 04:21:00, MorDt = 3840.0s
```
etc.

**Comment by RegMeasures** [comment]
![image](https://user-images.githubusercontent.com/17330572/77116034-4eebc600-6a94-11ea-99bf-dcbb749adfcd.png)
```
INFO   : Time = 2002-02-09 16:01:00, MorDt = 240.0s
INFO   : Time = 2002-02-09 17:03:00, MorDt = 240.0s
INFO   : Outlet channel elongated rightwards across transect line x=20.000000
INFO   : Preventing lagoonward truncation at X = 20.0 as it would result in a length 0 outlet channel
INFO   : Preventing lagoonward truncation means outlet channel pushed into cliff at X = 20.0
WARNING: Velocity change in cross-section 52 equals -140.267440 (greater than WarnTol)
...
WARNING: Velocity change in cross-section 21 equals -21168778.493914 (greater than WarnTol)
WARNING: Max iterations exceeded in unsteady river hydraulics - reverting to steady solution for this timestep
WARNING: Steady state solution results in critical depth at XS53
WARNING: Velocity change in cross-section 11 equals -8668.351555 (greater than WarnTol)
...
WARNING: Velocity change in cross-section 34 equals -43108936065651704.000000 (greater than WarnTol)
WARNING: Max iterations exceeded in unsteady river hydraulics - reverting to steady solution for this timestep
WARNING: Steady state solution results in critical depth at XS53
WARNING: Velocity change in cross-section 0 equals -229064990.142428 (greater than WarnTol)
WARNING: Velocity change in cross-section 24 equals 13731080079757328993259163222016.000000 (greater than WarnTol)
WARNING: LinAlgError singular matrix - adding some noise to see if this fixes it...
...
WARNING: LinAlgError singular matrix - adding some noise to see if this fixes it...
WARNING: Velocity change in cross-section 24 equals 13731080079757328993259163222016.000000 (greater than WarnTol)
WARNING: Max iterations exceeded in unsteady river hydraulics - reverting to steady solution for this timestep
WARNING: Steady state solution results in critical depth at XS53
WARNING: Velocity change in cross-section 0 equals -1711260291842962046103750251294570270510084877267805008822272.000000 (greater than WarnTol)
WARNING: LinAlgError singular matrix - adding some noise to see if this fixes it...
...
WARNING: LinAlgError singular matrix - adding some noise to see if this fixes it...
WARNING: Velocity change in cross-section 0 equals -1711260291842962046103750251294570270510084877267805008822272.000000 (greater than WarnTol)
WARNING: Max iterations exceeded in unsteady river hydraulics - reverting to steady solution for this timestep
WARNING: Steady state solution results in critical depth at XS53
WARNING: Depth change in cross-section 0 equals 9723046781305657008335877091579784944578848930482261852160.000000 (greater than WarnTol)
WARNING: Velocity change in cross-section 8 equals -95970690111188988993688849965365562480664521147742967246324517023952097982468283333901781021676855468178293536566366393335808.000000 (greater than WarnTol)
WARNING: array must not contain infs or NaNs
WARNING: Unsteady hydraulics failed at 02/09/2002, 18:03:00. Falling back to quasi-steady for this timestep.
WARNING: Steady state solution results in critical depth at XS53
INFO   : Lagoon overtopping barrier at X = 20.000000
INFO   : Outlet truncation due to breach at X = 20.000000
INFO   : Time = 2002-02-09 18:15:00, MorDt = 960.0s
```

**Comment by RegMeasures** [comment]
Need to handle eroded transects which do not cause breaches (see https://github.com/RegMeasures/HapuaModel/wiki/Notes-on-logic-rules-in-mor.updateMorphology) - will make a new issue for this

**Comment by RegMeasures** [comment]
Just found another example of lagoon backshore being pushed miles into cliff (@ X=-320) the only info messages at the time are:
```
INFO   : Outlet channel elongated leftwards across transect line X = -320.0m
INFO   : Preventing lagoonward truncation at X = -320.0m as it would result in a length 0 outlet channel
INFO   : Preventing lagoonward truncation means outlet channel pushed into cliff at X = -320.0m
```

**Comment by RegMeasures** [comment]
Seems to be caused by an earlier problem which creates an excessively high barrier crest height (`ShoreZ[:,0]`).

This problem appears to happen when:
```
INFO   : Overwashing occuring into closed channel in barrier - redistributing overwash onto barrier top
INFO   : Overwash into closed channel has leveled barrier at X = -320.0m. Removing channel from model.
```
